pkg/varlinkapi/transfers.go

```diff
@@ -0,0 +1,118 @@
+package varlinkapi
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/containers/libpod/cmd/podman/varlink"
+	"github.com/sirupsen/logrus"
+)
+
+// SendFile allows a client to send a file to the varlink server
+func (i *LibpodAPI) SendFile(call iopodman.VarlinkCall, ftype string, length int64) error {
+	if !call.WantsUpgrade() {
+		return call.ReplyErrorOccurred("client must use upgraded connection to send files")
+	}
+
+	outputFile, err := ioutil.TempFile("", "varlink_send")
+	if err != nil {
+		return call.ReplyErrorOccurred(err.Error())
+	}
+	defer outputFile.Close()
+
+	if err = call.ReplySendFile(outputFile.Name()); err != nil {
+		return call.ReplyErrorOccurred(err.Error())
+	}
+
+	writer := bufio.NewWriter(outputFile)
+	defer writer.Flush()
+
+	buf4096 := make([]byte, 4096)
+	var buf []byte
+
+	toBeCopied := int(length)
+
+	for toBeCopied > 0 {
```

Comment:
As above - there has to be something in the standard library that does this for us

Reply:
used copyn thanks!